sql_generators/glean_usage/templates/dau_reporting_clients_daily_v1.query.sql

```diff
@@ -0,0 +1,143 @@
+    DATE(parsed_end_time) AS session_end_date,
+  FROM
+    base
+),
+--
+with_date_offsets AS (
+  SELECT
+    *,
+    DATE_DIFF(submission_date, session_start_date, DAY) AS session_start_date_offset,
+    DATE_DIFF(submission_date, session_end_date, DAY) AS session_end_date_offset,
+  FROM
+    with_dates
+),
+--
+windowed AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    --
+    -- Take the earliest first_run_date if ambiguous.
+    MIN(first_run_date) OVER w1 AS first_run_date,
+    --
+    -- Sums over distinct dau_reporting pings.
+    -- SUM(IF(duration BETWEEN 0 AND 100000, duration, 0)) OVER w1 AS durations,
+    --
+    -- Bit patterns capturing activity dates relative to the submission date.
+    BIT_OR(
+      1 << IF(session_start_date_offset BETWEEN 0 AND 27, session_start_date_offset, NULL)
+    ) OVER w1 AS days_seen_session_start_bits,
+    BIT_OR(
+      1 << IF(session_end_date_offset BETWEEN 0 AND 27, session_end_date_offset, NULL)
+    ) OVER w1 AS days_seen_session_end_bits,
+    --
+    -- For all other dimensions, we use the mode of observed values in the day.
+    udf.mode_last(ARRAY_AGG(normalized_channel) OVER w1) AS normalized_channel,
+    udf.mode_last(ARRAY_AGG(normalized_os) OVER w1) AS normalized_os,
+    udf.mode_last(ARRAY_AGG(normalized_os_version) OVER w1) AS normalized_os_version,
+    udf.mode_last(ARRAY_AGG(locale) OVER w1) AS locale,
+    -- udf.mode_last(ARRAY_AGG(country) OVER w1) AS country,
+    udf.mode_last(ARRAY_AGG(app_build) OVER w1) AS app_build,
+    udf.mode_last(ARRAY_AGG(app_display_version) OVER w1) AS app_display_version,
+    udf.mode_last(ARRAY_AGG(distribution_id) OVER w1) AS distribution_id,
+    udf.mode_last(ARRAY_AGG(profile_group_id) OVER w1) AS profile_group_id,
+    {% if app_name == "firefox_desktop" %}
+    SUM(browser_engagement_uri_count) AS browser_engagement_uri_count,
+    SUM(browser_engagement_active_ticks) AS browser_engagement_active_ticks,
+    udf.mode_last(ARRAY_AGG(legacy_telemetry_client_id) OVER w1) AS legacy_telemetry_client_id,
+    {% endif %}
+  FROM
+    with_date_offsets
+  WHERE
+    {% raw %}
+    {% if is_init() %}
+      submission_date >= '2024-10-10'
+    {% else %}
+      submission_date = @submission_date
+    {% endif %}
+    {% endraw %}
+
+  WINDOW
+    w1 AS (
+      PARTITION BY
+        sample_id,
+        client_id,
+        submission_date
+      ORDER BY
+        submission_timestamp
+      ROWS BETWEEN
+        UNBOUNDED PRECEDING
+        AND UNBOUNDED FOLLOWING
+    ),
+    -- We must provide a modified window for ROW_NUMBER which cannot accept a frame clause.
+    w1_unframed AS (
+      PARTITION BY
+        sample_id,
+        client_id,
+        submission_date
+      ORDER BY
+        submission_timestamp
+    )
+),
+joined as (
+  SELECT
+    cd.* EXCEPT (_n),
+    cfs.first_seen_date,
```

Comment:
@gkatre @fbertsch is there really a good reason why we should have the first_seen_date in the daily table? The first_seen table is super minimal and if this information is necessarily it could be easily retrieved via a simple join. WDYT?

Reply:
Looking at the `clients_first_seen` query, i like the simplicity in which you have constructed the query. wonderful job!

My thoughts about having `first_seen_date` in the clients_daily table (but I could be overlooking some finer details):
I think it may not be needed in this table. I feel we need the computations of first_seen, last_seen and clients_daily only in the active_users table.

Also, looking at the `clients_daily` table, I am seeing that we are not filtering out `overactive` clients. Which means we could as well use the `clients_daily` table in the  `clients_first_seen` query for better performance over the `dau_reporting` ping table?